flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/savepoint/SavepointV0Serializer.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint.savepoint;
+
+import org.apache.flink.runtime.checkpoint.CompletedCheckpoint;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.OutputStream;
+
+/**
+ * Serializer for {@link SavepointV0} instances.
+ *
+ * <p>This format was introduced with Flink 1.0.0 and simply wraps a
+ * {@link CompletedCheckpoint} instance and uses plain Java serialization.
+ * That's why we have to stick to the exact same format, including classes,
+ * which are only part of our code base for backwards compatability.
+ */
+public class SavepointV0Serializer implements SavepointSerializer<SavepointV0> {
+
+	public static final SavepointV0Serializer INSTANCE = new SavepointV0Serializer();
```

Comment:
Maybe we should add a private constructor so that nobody can instantiate this class.